### PR TITLE
Simulate ANSI color escape codes on windows terminal

### PR DESCRIPTION
**Pull request by evilny0**
This is parsing the log output when it gets written, detecting ANSI color escape codes, and calling the relevant windows API function to simulate them.

Quite ugly hack I must say, but we get colors without reworking the log system.

If the windows terminal already supports ANSI codes, it will use ANSI codes and will not parse the output. However, I noticed the colors in powershell on Win10 are much better when NOT using ANSI codes. Dunno why, but maybe we should use the windows API even if the terminal supports ANSI ?

I could not really test it completely as I don't have a Windows system with a GPU, only a VM. However, it seems to work fine since I get colors on the error message saying I have no OpenCL device. Tested on Win10 cmd & powershell.

### CODE REVIEW

**Comment by Blacksuu** [comment]
Small feedback from win7, color kick in but there are 2 set of color:
A. when it start fresh first time miner
B. after it stop & start agen
in print screan you will see exactly what i say, it was done 2 printscrean & combine after in one single picture
http://i.imgur.com/has4WZI.png

**Comment by evilny0** [comment]
I see. It probably happens because I do not reset the console output when exiting, but only when instructed to by the log commands.

Depending on when the miner was stop, it might trigger this. I'll check it.

File: libdevcore/Terminal.h [hunk 1]
```diff
@@ -7,72 +7,72 @@ namespace con
...
-#define EthOrangeUnder	""       // Yellow
-#define EthNavyUnder	""       // Blue
-#define EthVioletUnder	""       // Purple
-#define EthTealUnder	""       // Cyan
+void HandleLogOutput(std::string const& _s);
+
+#define EthReset "\x1b[0m"       // Text Reset
+
+// Regular Colors
+#define EthBlack "\x1b[30m"        // Black
```
Comment (chfast) [inline libdevcore/Terminal.h]: Are the codes the same as on other systems? If yes, we don't need conditional `ifdef _WIN32`.

Comment (EoD) [inline libdevcore/Terminal.h]: Those should work on a ANSI/VT100 terminal on unix as well. I tried red, black and Reset on my unix terminal and they worked just fine.

Comment (evilny0) [inline libdevcore/Terminal.h]: Actually I just copied the Unix defines to Windows defines - I believe ANSI codes are the same on all systems.

**Comment by evilny0** [comment]
The issue with color sticking after stopping the process and restarting it should be fixed.

**Comment by chfast** [comment]
Still looks a bit messy to me, I will need a bit more time for it.

BTW, can you check https://github.com/ethereum-mining/ethminer/issues/197 ?

**Comment by XhmikosR** [comment]
Hi, guys. Any news about this?

**Comment by jean-m-cyr** [comment]
circleci seems to be broken!

**Comment by evilny0** [comment]
It's sometimes broken... here it's failing on getting ubuntu packages, so I don't think it's an issue :).

**Comment by jean-m-cyr** [comment]
Simpler solution provided by pr #698.

**Pull request closed without merging**